include/mfmg/dealii/belos_traits.hpp

```diff
@@ -0,0 +1,343 @@
+  //@{
+
+  /*! \brief Compute the 2-norm of each individual vector of \c mv.
+    Upon return, \c normvec[i] holds the value of \f$||mv_i||_2\f$, the \c i-th
+    column of \c mv.
+  */
+  static void
+  MvNorm(const MultiVectorType &mv,
+         std::vector<typename Teuchos::ScalarTraits<double>::magnitudeType>
+             &normvec,
+         NormType type = TwoNorm)
+  {
+    std::ignore = mv;
+    std::ignore = normvec;
+    std::ignore = type;
+    throw std::runtime_error("Not implemented");
+  }
+
+  //@}
+
+  //! @name Initialization methods
+  //@{
+  /*! \brief Copy the vectors in \c A to a set of vectors in \c mv indicated by
+  the indices given in \c index.
+
+  The \c numvecs vectors in \c A are copied to a subset of vectors in \c mv
+  indicated by the indices given in \c index, i.e.<tt> mv[index[i]] = A[i]</tt>.
+  */
+  static void SetBlock(const MultiVectorType &A, const std::vector<int> &index,
+                       MultiVectorType &mv)
+  {
+    std::ignore = A;
+    std::ignore = index;
+    std::ignore = mv;
+    throw std::runtime_error("Not implemented");
+  }
+
+  /// \brief Deep copy of A into specified columns of mv
+  ///
+  /// (Deeply) copy the first <tt>index.size()</tt> columns of \c A
+  /// into the columns of \c mv specified by the given index range.
+  ///
+  /// Postcondition: <tt>mv[i] = A[i - index.lbound()]</tt>
+  /// for all <tt>i</tt> in <tt>[index.lbound(), index.ubound()]</tt>
+  ///
+  /// \param A [in] Source multivector
+  /// \param index [in] Inclusive index range of columns of mv;
+  ///   index set of the target
+  /// \param mv [out] Target multivector
+  static void SetBlock(const MultiVectorType &A, const Teuchos::Range1D &index,
+                       MultiVectorType &mv)
+  {
+    std::ignore = A;
+    std::ignore = index;
+    std::ignore = mv;
+    throw std::runtime_error("Not implemented");
+  }
+
+  /// \brief mv := A
+  ///
+  /// Assign (deep copy) A into mv.
+  static void Assign(const MultiVectorType &A, MultiVectorType &mv)
+  {
+    std::ignore = A;
+    std::ignore = mv;
+    throw std::runtime_error("Not implemented");
+  }
+
+  /*! \brief Replace the vectors in \c mv with random vectors.
+   */
+  static void MvRandom(MultiVectorType &mv)
+  {
+    std::ignore = mv;
+    throw std::runtime_error("Not implemented");
+  }
+
+  /*! \brief Replace each element of the vectors in \c mv with \c alpha.
+   */
+  static void MvInit(MultiVectorType &mv, const double alpha = 0.)
+  {
+    std::ignore = mv;
+    std::ignore = alpha;
+    throw std::runtime_error("Not implemented");
+  }
+
+  /*! \brief Print the \c mv multi-vector to the \c os output stream.
+   */
+  static void MvPrint(const MultiVectorType &mv, std::ostream &os)
+  {
+    std::ignore = mv;
+    std::ignore = os;
+    throw std::runtime_error("Not implemented");
+  }
+
+  static bool HasConstantStride(const MultiVectorType &mv)
+  {
+    std::ignore = mv;
+    throw std::runtime_error("Not implemented");
+  }
```

Comment:
If you don't implement anything why do you even have this file?

Reply:
Because Anasazi :( Some solvers in it use Belos, and they are instantiated if Belos is enabled. Those instantiations require one to also provide Belos traits. Generalized Davidson or LOBPCG don't need it, so I don't actually have to implement it, just to provide so that things compile.